src/html-imports.js

```diff
@@ -498,29 +509,22 @@
         // Check if was already moved to head, to handle the case where the element
         // has already been moved but it is still loading.
         if (needsMove && s.parentNode !== document.head) {
-          let rootImport = importForElement(s);
-          while (rootImport && importForElement(rootImport)) {
-            rootImport = importForElement(rootImport);
-          }
           // Replace the element we're about to move with a placeholder.
-          // NOTE: we first have to append the element to the new parent, then
-          // we can put the placeholder at its place, otherwise load/error events
-          // seem to be fired too early.
-          const parent = s.parentNode,
-            next = s.nextSibling,
-            placeholder = document.createElement(s.localName);
+          const placeholder = document.createElement(s.localName);
           // Add reference of the moved element.
           placeholder['__appliedElement'] = s;
           // Disable this from appearing in document.styleSheets.
           placeholder.setAttribute('type', 'import-placeholder');
-          // First, re-parent the element...
-          if (rootImport.parentNode === document.head) {
-            document.head.insertBefore(s, rootImport);
-          } else {
-            document.head.appendChild(s);
+          // Append placeholder next to the sibling, and move original to <head>.
+          s.parentNode.insertBefore(placeholder, s.nextSibling);
+          let newSibling = importForElement(s);
+          while (newSibling && importForElement(newSibling)) {
+            newSibling = importForElement(newSibling);
```

Comment:
`importForElement` returns the memoized importDoc if available, so calling the same method again shouldn't be expensive.
To avoid calling this method again I'd have to use a temporary variable, is that preferable?